src/categorical_algebra/ACSetViews.jl

```diff
@@ -0,0 +1,185 @@
+    attr_types = map(a -> Ts.parameters[codom_num(AD,a)], attr_names)
+    row_type = NamedTuple{attr_names, Tuple{attr_types...}}
+    new{A,P,row_type}(backing,parts)
+  end
+  function ACSetView(backing::A, P::Symbol) where
+      {CD <: CatDesc,AD <: AttrDesc{CD},Ts <: Tuple, A <: ACSet{CD,AD,Ts}}
+    ACSetView(backing,P,1:nparts(backing,P))
+  end
+end
+
+# We're going to overload getproperty, so we use these instead
+
+backing(av::ACSetView) = getfield(av,:backing)
+
+CSetDataStructures.parts(av::ACSetView) = getfield(av,:parts)
+
+
+# Overloaded accessors
+######################
+
+Base.size(av::ACSetView) = size(parts(av))
+
+@generated function Base.getindex(av::ACSetView{<:ACSet,P,Attrs}, i::Int) where
+    {P, Attrs <: NamedTuple}
+  tuple_args = map(Attrs.parameters[1]) do attr
+    :($attr = av[i,$(Expr(:quote, attr))])
+  end
+  Expr(:tuple,tuple_args...)
+end
+
+function Base.getindex(av::ACSetView{<:ACSet,P,<:NamedTuple},i::Int,name::Union{Symbol,Vector{Symbol}}) where {P}
+  subpart(backing(av), parts(av)[i], name)
+end
+
+function Base.getindex(av::ACSetView{<:ACSet,P,<:NamedTuple},i,name::Union{Symbol,Vector{Symbol}}) where {P}
+  subpart(backing(av), view(parts(av),i), name)
+end
+
+function Base.getindex(av::ACSetView{<:ACSet,P,<:NamedTuple},name::Union{Symbol,Vector{Symbol}}) where {P}
+  subpart(backing(av), parts(av), name)
+end
+
+function Base.getindex(av::ACSetView{<:ACSet,P,<:NamedTuple},indicator::BitArray{1}) where {P}
+  ACSetView(backing(av),P,parts(av)[indicator])
+end
+
+Base.getproperty(av::ACSetView, name::Symbol) =
+  subpart(backing(av), parts(av), name)
+
+function Base.show(io::IO, ::MIME"text/plain", av::T) where {ACS <: AbstractACSet, P, T<:ACSetView{ACS,P}}
+  print(io, "ACSetView of object $P")
+  println(io)
+  table = tables(backing(av))[P][parts(av)]
+  if !(eltype(table) <: NamedTuple{(),Tuple{}} || isempty(table))
+    pretty_table(io, table, nosubheader=true,
+                 show_row_number=true, row_number_column_title=string(P))
+  end
+end
+
+# Tables.jl interface
+#####################
+
+Tables.istable(av::ACSetView) = true
+
+Tables.columnaccess(av::ACSetView) = true
+Tables.columns(av::ACSetView) = av
+
+function Tables.getcolumn(av::ACSetView{<:ACSet,P,Attrs}, i::Int) where {P, Attrs <: NamedTuple}
+  av[:,Attrs.parameters[1][i]]
+end
+
+Tables.getcolumn(av::ACSetView, nm::Symbol) = av[:,nm]
+
+Tables.getcolumn(av::ACSetView, i::Int, nm::Symbol) = av[i,nm]
+
+Tables.columnnames(av::ACSetView{<:ACSet,P,Attrs}) where {P,Attrs<:NamedTuple} = Attrs.parameters[1]
+
+# Macros for Querying/filtering
+###############################
+
+""" This macro takes an expression using attributes of the object selected by the ACSetView,
+and returns an array of the result of that expression applied to the attribute of each part
+in the ACSetView.
+
+Example: @compute_prop(edges_table, 2 * src.weight)
+
+Eventually, this should be superceded by Query.jl or AlgebraicRelations.jl functionality, so I have
```

Comment:
Lol at "purposely made the name a bit obtuse and awkward"!

The ability to use compound attributes is nice. IMO, if you think this is useful (it seems useful to me!), you should give it a worthy name.

Reply:
I think this function is useful, but I think that it should be replaced by something more useful. And actually, I think `@compute_prop` is not so bad.